### ISSUE TITLE
 E0428 does not detected for functions marked as test

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8409]
## Environment

* **IntelliJ Rust plugin version:** 0.4.164.4364-213-nightly
* **Rust toolchain version:** 1.57.0 (f1edd0429 2021-11-29) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3.2 (CL-213.6461.75)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
No error shown for fn `func`:
<img width="682" alt="изображение" src="https://user-images.githubusercontent.com/49211026/149512589-21c2c409-4c68-47f4-89ff-cfd9c8d3c626.png">
Changing order of function definition changes situation, but still incorrect:
<img width="285" alt="изображение" src="https://user-images.githubusercontent.com/49211026/149512772-e0d045b4-09d4-421b-9d8c-005569987b83.png">

It seems like it ignores the first function.

After removing all `#[test]` errors are shown for both names.

## Steps to reproduce
in lib.rs
```rust
#[cfg(test)]

#[test]
fn kva() {
    let result = 2 + 2;
    assert_eq!(result, 4);
}

#[test]
fn func() {
    let result = 2 + 2;
    assert_eq!(result, 4);
}

#[test]
fn func() {
    let result = 2 + 2;
    assert_eq!(result, 4);
}

#[test]
fn kva() {
    let result = 2 + 2;
    assert_eq!(result, 4);
}

```
